crates/re_log_encoding/src/decoder.rs

```diff
@@ -23,12 +39,23 @@ fn warn_on_version_mismatch(encoded_version: [u8; 4]) {
 
     const LOCAL_VERSION: CrateVersion = CrateVersion::parse(env!("CARGO_PKG_VERSION"));
 
-    if !encoded_version.is_compatible_with(LOCAL_VERSION) {
-        re_log::warn!(
-            "Found log stream with Rerun version {encoded_version}, \
-            which is incompatible with the local Rerun version {LOCAL_VERSION}. \
-            Loading will try to continue, but might fail in subtle ways."
-        );
+    if encoded_version.is_compatible_with(LOCAL_VERSION) {
+        Ok(())
+    } else {
+        match version_policy {
+            VersionPolicy::Warn => {
+                re_log::warn!(
+                    "Found log stream with Rerun version {encoded_version}, \
+                     which is incompatible with the local Rerun version {LOCAL_VERSION}. \
+                     Loading will try to continue, but might fail in subtle ways."
+                );
+                Ok(())
+            }
+            VersionPolicy::Error => Err(DecodeError::IncompatibleRerunVersion {
+                file: encoded_version,
+                local: LOCAL_VERSION,
+            }),
+        }
```

Comment:
Any reason not to print the error in the warn case? Is it looking worse?
```suggestion
        let err = DecodeError::IncompatibleRerunVersion {
                file: encoded_version,
                local: LOCAL_VERSION,
        };
        match version_policy {
            VersionPolicy::Warn => {
                re_log::warn!("{}", err);
                Ok(())
            }
            VersionPolicy::Error => Err(err),
       }
```

Reply:
One major difference is "Loading will try to continue, but might fail in subtle ways.".

The other is that the error might print something about it being a failure, but the warning should not